### PR TITLE
FEATURE : Filter tasks by  user and title

### PR DESCRIPTION
**Pull request by sahsisunny**
### Description:
Added functionality to search tasks by assignee and title.

### Anything you would like to inform the reviewer about:
none

### Dev Tested:
- [x] Yes

### Test Cases
image
image
image
image
image
image

### Images/video of the change:

https://github.com/Real-Dev-Squad/website-status/assets/70854507/3359c392-580e-4e96-acf2-2e8470cfc220

### CODE REVIEW

File: __tests__/Unit/Components/Tasks/TasksContent.test.tsx [hunk 1]
```diff
@@ -110,18 +109,14 @@ describe('tasks content', () => {
         const tabsContainer = within(
             screen.getByTestId('status-tabs-container')
         );
-        const inProgressBtn = tabsContainer.getByRole('button', {
-            name: /IN PROGRESS/i,
-        });
-        expect(inProgressBtn).toHaveClass('active');
```
Comment (vinit717) [inline __tests__/Unit/Components/Tasks/TasksContent.test.tsx]: Please don't remove this test

File: __tests__/Unit/hooks/tasksApi.test.tsx
Comment (vinit717) [inline __tests__/Unit/hooks/tasksApi.test.tsx:1]: Please add a test for the wrong title and empty title

File: src/components/tasks/TasksContent.tsx [hunk 2]
```diff
@@ -10,23 +10,31 @@ import {
...
 
-const getQueryParamValue = (tab: Tab) => `is:${tabToUrlParams(tab)}`;
-
 export const TasksContent = ({ dev }: { dev: boolean }) => {
     const router = useRouter();
-    const allQueryParams = router.query;
-    const qQueryParam = allQueryParams.q as string;
-    const queryTab = qQueryParam?.replace('is:', '');
-    const selectedTab = getActiveTab(queryTab);
+    const qQueryParam = router.query.q as string;
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply src/components/tasks/TasksContent.tsx:28]: I believe that using "qQueryParam" as the identifier for the "q" query parameter is a suitable choice.

**Review by vinit717 — APPROVED** [review]

**Review by shreya-mishra — APPROVED** [review]
